egs/wsj/s5/steps/segmentation/internal/prepare_sad_graph.py

```diff
@@ -0,0 +1,164 @@
+
+sys.path.insert(0, 'steps')
+import libs.common as common_lib
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler()
+handler.setLevel(logging.INFO)
+formatter = logging.Formatter("%(asctime)s [%(filename)s:%(lineno)s - "
+                              "%(funcName)s - %(levelname)s ] %(message)s")
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+
+def get_args():
+    parser = argparse.ArgumentParser(
+        description="This script generates a graph directory for decoding with "
+        "a simple HMM model.\n"
+        "It needs as an input classes_info file with the format:\n"
+        "<class-id (1-indexed)> <initial-probability> <self-loop-probability> "
+        "<min-duration> <list-of-pairs>,\n"
+        "where each pair is <destination-class>:<transition-probability>.\n"
+        "destination-class -1 is used to represent final probability.",
+        formatter_class=argparse.RawTextHelpFormatter)
+
+    parser.add_argument("--transition-scale", type=float, default=1.0,
+                        help="""Scale on transition probabilities relative to
+                        LM weights""")
+    parser.add_argument("--loopscale", type=float, default=0.1,
+                        help="""Scale on self-loop log-probabilities relative
+                        to LM weights""")
+
+    parser.add_argument("--min-silence-duration", type=float, default=0.03,
+                        help="""Minimum duration for silence""")
+    parser.add_argument("--min-speech-duration", type=float, default=0.3,
+                        help="""Minimum duration for speech""")
+    parser.add_argument("--max-speech-duration", type=float, default=10.0,
+                        help="""Maximum duration for speech""")
+    parser.add_argument("--frame-shift", type=float, default=0.03,
+                        help="""Frame shift in seconds""")
+
+    parser.add_argument("--edge-silence-probability", type=float,
+                        default=0.5,
+                        help="Probability of silence at the edges.")
+    parser.add_argument("--transition-probability", type=float, default=0.1,
+                        help="Transition probability for silence to speech "
+                        "or vice-versa")
+
+    parser.add_argument("output_graph", type=str,
+                        help="Output graph")
+    args = parser.parse_args()
+
+    args.min_states_silence = int(args.min_silence_duration / args.frame_shift
+                                  + 0.5)
+    args.min_states_speech = int(args.min_speech_duration / args.frame_shift
+                                 + 0.5)
+    args.max_states_speech = int(args.max_speech_duration / args.frame_shift
+                                 + 0.5)
+
+    return args
+
+
+def print_states(args, file_handle):
+    # Initial transition to silence
+    print ("0 1 1 1 {0}".format(-math.log(args.edge_silence_probability)),
+           file=file_handle)
```

Comment:
An option: since the graph is acceptor, do not duplicate the label and then `fstcompile` with the  `--acceptor` command line switch.